### PR TITLE
fix: reduce retryprovider env variables

### PR DESCRIPTION
**Pull request by amateima**
- adds default values to many of the RetryProvider ENV variables
- implement a new `RetryProvidersFactory` class for instantiating the RetryProviders and using a single instance per chain

### CODE REVIEW

File: packages/indexer/src/parseEnv.ts [hunk 1]
```diff
@@ -115,6 +117,68 @@ function parseRetryProviderConfig(
...
+      "0",
+  );
+  const providerCacheTtl = process.env.PROVIDER_CACHE_TTL
+    ? Number(process.env.PROVIDER_CACHE_TTL)
+    : undefined;
+  const nodeQuorumThreshold = Number(
+    process.env[`NODE_QUORUM_${chainId}`] || process.env.NODE_QUORUM || "1",
+  );
+  const retries = Number(
+    process.env[`NODE_RETRIES_${chainId}`] || process.env.NODE_RETRIES || "2",
```
Comment (daywiss) [inline packages/indexer/src/parseEnv.ts:155]: id probably set this to 1 or 0 (whatever no retries is) if you are omitting this

**Review by daywiss — APPROVED** [review]
looks good

File: packages/indexer/src/web3/RetryProvidersFactory.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  constructor(
+    private redisCache: RedisCache,
+    private logger: Logger,
+  ) {}
+
+  public initializeProviders() {
+    const providersUrls = parseProvidersUrls();
+
+    for (const chainId of this.retryProviders.keys()) {
```
Comment (melisaguevara) [inline packages/indexer/src/web3/RetryProvidersFactory.ts:19]: Should this use `providersUrls` keys instead?

**Review by melisaguevara — APPROVED** [review]

File: packages/indexer/src/web3/constants.ts
Comment (james-a-morris) [inline packages/indexer/src/web3/constants.ts:1]: Are these meant to be largely unchanged as time goes on?

**Review by james-a-morris — APPROVED** [review]
one question

File: packages/indexer/src/web3/RetryProvidersFactory.ts (hunk 2 above)
Comment (amateima) [reply packages/indexer/src/web3/RetryProvidersFactory.ts:19]: Yes! Good catch! I actually fixed this in my upcoming PR but I forgot to fix it on the branch I was branching of 😄

File: packages/indexer/src/web3/constants.ts
Comment (amateima) [reply packages/indexer/src/web3/constants.ts:1]: Supposedly yes

**Pull request merged**